modules/fedimint-ln-client/src/lib.rs

```diff
@@ -838,26 +838,29 @@ impl LightningClientModule {
             Some(active_gateway) => Ok(active_gateway.info),
             None => {
                 let gateways = self.fetch_registered_gateways().await?;
+                let num_gateways = gateways.len();
 
                 let vetted = gateways
                     .clone()
                     .into_iter()
                     .filter(|g| g.vetted)
                     .collect::<Vec<_>>();
                 if !vetted.is_empty() {
-                    debug!("Choosing a vetted gateway");
-                    vetted
+                    let chosen_gateway = vetted
                         .into_iter()
                         .map(|gw| gw.info)
                         .choose(&mut rand::thread_rng())
-                        .ok_or(anyhow::anyhow!("Could not choose a vetted gateway"))
+                        .expect("vetted is non-empty");
+                    debug!(gateway_id = %chosen_gateway.gateway_id, %num_gateways, "Choosing a vetted gateway");
```

Comment:
you don't use display for primitive types - https://docs.rs/tracing/latest/tracing/trait.Value.html#foreign-impls
```suggestion
                    debug!(gateway_id = %chosen_gateway.gateway_id, num_gateways, "Choosing a vetted gateway");
```

Reply:
Thx, updated!